debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java

```diff
@@ -2881,6 +2885,83 @@ public void testStreamingWithNumericReplicationSlotName() throws Exception {
         assertInsert(recordsForTopic.get(1), PK_FIELD, 201);
         assertInsert(recordsForTopic.get(2), PK_FIELD, 202);
         assertInsert(recordsForTopic.get(3), PK_FIELD, 203);
+
+    @Test
+    @FixFor("DBZ-2363")
+    @SkipWhenDecoderPluginNameIsNot(value = SkipWhenDecoderPluginNameIsNot.DecoderPluginName.PGOUTPUT, reason = "Only supported on PgOutput")
+    @SkipWhenDatabaseVersion(check = LESS_THAN, major = 14, minor = 0, reason = "Database Version less than 14")
+    public void shouldNotConsumeLogicalDecodingMessagesWhenConfigIsSetToSkip() throws Exception {
+        TestHelper.execute(SETUP_TABLES_STMT);
+        Configuration.Builder configBuilder = TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.LOGICAL_DECODING_MESSAGE_HANDLING_MODE, "skip");
+        start(PostgresConnector.class, configBuilder.build());
+        assertConnectorIsRunning();
+        waitForSnapshotToBeCompleted();
+
+        // emit logical decoding message
+        TestHelper.execute("SELECT pg_logical_emit_message(false, 'prefix', 'content');");
+
+        assertNoRecordsToConsume();
+    }
+
+    @Test
+    @FixFor("DBZ-2363")
+    @SkipWhenDecoderPluginNameIsNot(value = SkipWhenDecoderPluginNameIsNot.DecoderPluginName.PGOUTPUT, reason = "Only supported on PgOutput")
+    @SkipWhenDatabaseVersion(check = LESS_THAN, major = 14, minor = 0, reason = "Database Version less than 14")
+    public void shouldConsumeLogicalDecodingMessagesWhenConfigIsSetToInclude() throws Exception {
+        TestHelper.execute(SETUP_TABLES_STMT);
+        Configuration.Builder configBuilder = TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.LOGICAL_DECODING_MESSAGE_HANDLING_MODE, "include");
+
+        start(PostgresConnector.class, configBuilder.build());
+        assertConnectorIsRunning();
+
+        assertRecordsFromSnapshot(2, 1, 1);
+
+        // emit non transactional logical decoding message
+        TestHelper.execute("SELECT pg_logical_emit_message(false, 'foo', 'bar');");
+
+        SourceRecords records = consumeRecordsByTopic(1);
+        List<SourceRecord> recordsForTopic = records.recordsForTopic(topicName("logical_decoding_message"));
```

Comment:
How is the topic name determined actually? This should contain the "logical server name" of the connector (the "dbserver1" prefix in many of the tests/examples). Perhaps `<logical name>.message` would make sense (i.e. omitting the implementation detail of this being based on logical decoding)?

Reply:
I like that. The topic suffix is set in the `LogicalDecodingMessageMonitor`  similar to the `TransactionMonitor`. I can update that to be `message`. https://github.com/debezium/debezium/blob/89eb71032167b15906e2a78fed14f9500a56e982/debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/LogicalDecodingMessageMonitor.java#L48-L56

Reply:
Cool, thanks!